### PR TITLE
Near/borsh wrapper

### PR DESCRIPTION
**Pull request by Brioline**

### CODE REVIEW

File: protocol/near/borsh-wrapper/src/__tests__/fuzz/borsh.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+const {
+  ensAddresses,
+  initTestEnvironment,
+  providers,
+} = require("@polywrap/test-env-js");
+const { PolywrapClient } = require("@polywrap/client-js");
+const testUtils = require("../testUtils");
```
Comment (krisbitney) [inline protocol/near/borsh-wrapper/src/__tests__/fuzz/borsh.ts:7]: This file was renamed to `utils.ts`

Comment (krisbitney) [reply protocol/near/borsh-wrapper/src/__tests__/fuzz/borsh.ts:7]: The fuzzing isn't working on my local machine. I'm not sure why. 
I get:
`Error: Cannot find module '../testUtils'`
I tried updating the import and I still get:
`Error: Cannot find module '../utils'`

**Review by krisbitney — APPROVED** [review]
Amazing work!

I'm curious, did you see https://github.com/gagdiez/serial-as/tree/main/borsh? What did you think?

Note for Polywrap: The Near interface doesn't build because it requires a `Module` type to build. Should interfaces require a `Module`?

File: protocol/near/borsh-wrapper/src/__tests__/fuzz/borsh.ts (hunk 1 above)
Comment (Brioline) [reply protocol/near/borsh-wrapper/src/__tests__/fuzz/borsh.ts:7]: Hi @krisbitney ! Borsh serializer you provided link for, is a straightforward encoder/decoder of simple values. Near uses typescript borsh implementation which uses schema to serialize/deserialize custom classes, which produce some custom encodings.
Also, there is a difference in enocoding/decoding some values between borsh-as and borsh-ts. Let's take an Array for example. TS version just stores bytes to encoder, AS version writes length first, then bytes - it results in incorrect bytes, and near backend does not recognize them. This Borsh-as implementation mimics borsh-ts transaction creatione to make exact same bytes to be valid

**Pull request merged**